packages/plugin-survey/src/index.ts

```diff
@@ -165,165 +263,411 @@ class SurveyPlugin implements JsPsychPlugin<Info> {
-    survey.completeText = trial.button_label_finish;
+    this.survey.pagePrevText = this.params.button_label_back;
+    this.survey.pageNextText = this.params.button_label_next;
+    this.survey.completeText = this.params.button_label_finish;
 
     // page numbers
-    survey.showQuestionNumbers = trial.show_question_numbers;
+    this.survey.showQuestionNumbers = this.params.show_question_numbers;
 
     // survey title
-    if (trial.title !== null) {
-      survey.title = trial.title;
+    if (this.params.title !== null) {
+      this.survey.title = this.params.title;
     }
 
+    // required question label
+    this.survey.requiredText = this.params.required_question_label;
+
+    // TO DO: response validation
+    this.survey.checkErrorsMode = "onNextPage"; // onValueChanged
+
+    // TO DO: automatic response accuracy scoring for questions with a correctAnswer value
+    // see Survey Model onIsAnswerCorrect event
+    // https://surveyjs.io/Documentation/Library/?id=SurveyModel#onIsAnswerCorrect
+
     // pages and questions
-    for (let i = 0; i < trial.pages.length; i++) {
+    for (let i = 0; i < this.params.pages.length; i++) {
       let page_id = "page" + i.toString();
-      let page = survey.addNewPage(page_id);
-      if (trial.randomize_question_order) {
+      let page = this.survey.addNewPage(page_id);
+      if (this.params.randomize_question_order) {
         page.questionsOrder = "random"; // TO DO: save question presentation order to data
       }
-      for (let j = 0; j < trial.pages[i].length; j++) {
-        let question_params = trial.pages[i][j];
-        let q_type = this.question_type_map[question_params.type];
+      for (let j = 0; j < this.params.pages[i].length; j++) {
+        let question_params = this.params.pages[i][j];
+        // question type validation
+        let q_type: string;
+        let q_opts: readonly string[] = Object.keys(this.question_type_map);
+        if (typeof question_params.type == "undefined") {
+          throw new Error(
+            'Error in survey plugin: question is missing the required "type" parameter.'
+          );
+        } else if (!q_opts.includes(question_params.type)) {
+          throw new Error(
+            'Error in survey plugin: invalid question type "' + question_params.type + '".'
+          );
+        } else {
+          q_type = this.question_type_map[question_params.type];
+        }
         if (q_type == "text" && question_params.textbox_rows > 1) {
           q_type = "comment";
         }
+        // set up question
         let question = page.addNewQuestion(q_type);
-        question.name = "p" + i.toString() + "_q" + j.toString();
+        question.requiredErrorText = this.params.required_error_text;
+        if (typeof question_params.name !== "undefined") {
+          question.name = question_params.name;
+        } else {
+          question.name = "P" + i.toString() + "_Q" + j.toString();
+        }
         switch (q_type) {
```

Comment:
If you'd like to save some lines, you could have an object that maps strings to setup functions here